main.go

```diff
@@ -0,0 +1,225 @@
+
+	err := json.Unmarshal(data, &objects) //unmarshalling
+	if err != nil {
+		return err
+	}
+	//then we normally unmarshal the data into all the Train struct variables except for ArrivalTime and Departure time.
+	err = json.Unmarshal(*objects["trainId"], &t.TrainID)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(*objects["departureStationId"], &t.DepartureStationID)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(*objects["arrivalStationId"], &t.ArrivalStationID)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(*objects["price"], &t.Price)
+	if err != nil {
+		return err
+	}
+	var Arr, Dep string //we use string variables to get the lines from the json file
+	err = json.Unmarshal(*objects["arrivalTime"], &Arr)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(*objects["departureTime"], &Dep)
+	if err != nil {
+		return err
+	}
+	var h, m, s int
+	_, err = fmt.Sscanf(Arr, "%d:%d:%d", &h, &m, &s) //then we convert them into the varaibles h(ours), m(inutes),and s(econds)
+	if err != nil {
+		return err
+	}
+	t.ArrivalTime = time.Date(0, time.January, 1, //and then we set the Arrival and Departure time!
+		h, m, s, 0,
+		time.UTC)
+	_, err = fmt.Sscanf(Dep, "%d:%d:%d", &h, &m, &s)
+	if err != nil {
+		return err
+	}
+	t.DepartureTime = time.Date(0, time.January, 1,
+		h, m, s, 0,
+		time.UTC)
+	return nil
+}
+func main() {
+
+	byteValue, err := ioutil.ReadFile("data.json") //Opening file
+	if err != nil {
+		log.Fatal("Error when opening file:", err)
+	}
+
+	var trains Trains
+
+	err = json.Unmarshal(byteValue, &trains) //Unmarshalling the file into the variable trains
+	if err != nil {
+		log.Fatal("Error during Unmarshal(): ", err)
+	}
+
+	stationDepNums := make(map[int][]int) //maps we'll use to find trains that match our departure and arrival stations
+	stationArrNums := make(map[int][]int)
+
+	for i := 0; i < len(trains); i++ {
+		stationDepNums[trains[i].DepartureStationID] = append(stationDepNums[trains[i].DepartureStationID], i) //adds trains to matching maps
+		stationArrNums[trains[i].ArrivalStationID] = append(stationArrNums[trains[i].ArrivalStationID], i)     //i.e if DepartureStationID = 10 in a Train varaible, the location of the train within Trains gets added to a slice in stationDepNums corresponding to the key "10"
+	}
+
+	var departureStation string
+	var arrivalStation string
+	var criteria string
+
+	fmt.Println("Enter departure station:") //departure
+	fmt.Scanln(&departureStation)
+
+	if departureStation == "" {
+		log.Fatal(errors.New("empty departure station"))
+	}
+	if _, err := strconv.Atoi(departureStation); err != nil {
+		log.Fatal(errors.New("bad departure station input"))
+	}
+
+	fmt.Println("Enter arrival station:") //arrival
+	fmt.Scanln(&arrivalStation)           //Yes, i know that the varaible named "departureStation" is used to get the arrival station ID. this is because i noticed that
```

Comment:
Ooops. I made the comment when there was a weird bug that mixed up arrivalStation and departureStation, i fixed it but forgot to remove the comment.